File: comms/push-notifications/list-push-notifications/list-push-notifications.cs
using Twilio.Comms;
using System.Threading.Tasks;
using System;
using System.Globalization;

public partial class Examples
{
    public async Task Example() {
        var client = new TwilioComms(
            accountId: "TWILIO_ACCOUNT_SID",
            authToken: "TWILIO_AUTH_TOKEN"
        );

        await client.PushNotifications.ListAsync(
            new PushNotificationsListRequest {
                OperationId = "comms_operation_01h9krwprkeee8fzqspvwy6nq8",
                Status = PushNotificationStatus.Scheduled,
                Tags = "key_1:value;key_2:value;",
                StartDate = DateTime.Parse("2024-01-15T09:30:00Z", null, DateTimeStyles.AdjustToUniversal),
                EndDate = DateTime.Parse("2024-01-15T09:30:00Z", null, DateTimeStyles.AdjustToUniversal),
                User = "comms_pushnotificationuser_01h9krwprkeee8fzqspvwy6nq8",
                Provider = PushNotificationProvider.Apn,
                PageToken = "pageToken",
                PageSize = 50
            }
        );
    }

}

File: comms/push-notifications/list-push-notifications/list-push-notifications.rb
require "twiliocomms"

client = TwilioComms::Client.new(
  account_id: "TWILIO_ACCOUNT_SID",
  auth_token: "TWILIO_AUTH_TOKEN"
)

client.push_notifications.list(
  operation_id: "comms_operation_01h9krwprkeee8fzqspvwy6nq8",
  status: "SCHEDULED",
  tags: "key_1:value;key_2:value;",
  start_date: "2024-01-15T09:30:00Z",
  end_date: "2024-01-15T09:30:00Z",
  user: "comms_pushnotificationuser_01h9krwprkeee8fzqspvwy6nq8",
  provider: "APN",
  page_token: "pageToken",
  page_size: 50
)
